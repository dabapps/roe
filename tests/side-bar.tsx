import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SideBar from '../src/ts/components/navigation/side-bar';

describe('SideBar', () => {

  it('should match snapshot', () => {
    const tree = renderer.create(
      <SideBar
        position="left"
        onClickOutside={jest.fn()}
      />
    );

    expect(tree).toMatchSnapshot();
  });

  it('should accept regular element attributes', () => {
    const tree = renderer.create(
      <SideBar
        className="my-class"
        position="left"
        onClickOutside={jest.fn()}
      />
    );

    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot when open', () => {
    const tree = renderer.create(
      <SideBar
        open
        position="left"
        onClickOutside={jest.fn()}
      />
    );

    expect(tree).toMatchSnapshot();
  });

});
