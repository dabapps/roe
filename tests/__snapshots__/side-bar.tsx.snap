// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should accept a component props 1`] = `
<div
  className="side-bar-container"
>
  <span />
  <ul
    className="side-bar left"
  />
</div>
`;

exports[`SideBar should accept regular element attributes 1`] = `
<div
  className="side-bar-container my-class"
>
  <span />
  <div
    className="side-bar left"
  />
</div>
`;

exports[`SideBar should apply no-shadow class 1`] = `
<div
  className="side-bar-container"
>
  <span />
  <div
    className="side-bar no-shadow left"
  />
</div>
`;

exports[`SideBar should match snapshot 1`] = `
<div
  className="side-bar-container"
>
  <span />
  <div
    className="side-bar left"
  />
</div>
`;

exports[`SideBar should match snapshot right 1`] = `
<div
  className="side-bar-container"
>
  <span />
  <div
    className="side-bar right"
  />
</div>
`;

exports[`SideBar should match snapshot when open 1`] = `
<div
  className="side-bar-container"
>
  <span>
    <div
      className="side-bar-overlay"
      onClick={[MockFunction]}
    />
  </span>
  <div
    className="side-bar left open"
  />
</div>
`;
